vllm/entrypoints/openai/serving_engine.py

```diff
@@ -953,26 +954,32 @@ async def _normalize_prompt_text_to_input(
             prompt = prompt.lower()
 
         truncate_prompt_tokens = getattr(request, "truncate_prompt_tokens", None)
+        # When explicitly set to 0 or False, honor the caller and skip truncation
+        disable_truncation = (
+            truncate_prompt_tokens == 0 or truncate_prompt_tokens is False
+        )
```

Comment:
Suggest to apply truncation in all cases here. It should not be possible to disable.